### PR TITLE
Mic-5004/risk-effect-config

### PR DESCRIPTION
**Pull request by albrja**
### Adds configurable data lookup tables for RiskEffect class
- *Category*: Feature
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5004

### Changes and notes
-updates configuration in Risk for PAF and relative risk
-updates RiskEffect to for configuration changes to create lookup tables

### Testing
All tests pass

### CODE REVIEW

File: src/vivarium_public_health/risks/data_transformations.py [hunk 1]
```diff
@@ -393,25 +397,31 @@ def exposure_effect(rates, rr: pd.DataFrame) -> pd.Series:
...
+                "population_attributable_fraction"
+            ],
         )
         correct_target = (paf_data["affected_entity"] == target.name) & (
             paf_data["affected_measure"] == target.measure
         )
+        # TODO: what to do about this?
         paf_data = paf_data[correct_target].drop(
             columns=["affected_entity", "affected_measure"]
         )
```
Comment (rmudambi) [inline src/vivarium_public_health/risks/data_transformations.py:422]: This all needs to be in an if block like so:
```python
if isinstance(paf_data, pd.DataFrame):
    ...
```

File: src/vivarium_public_health/risks/effect.py [hunk 2]
```diff
@@ -100,10 +103,10 @@ def __init__(self, risk: str, target: str):
     def setup(self, builder: Builder) -> None:
         self.exposure_distribution_type = self.get_distribution_type(builder)
         self.exposure = self.get_risk_exposure(builder)
-        self.relative_risk = self.get_relative_risk_source(builder)
-        self.population_attributable_fraction = (
-            self.get_population_attributable_fraction_source(builder)
-        )
+        # self.relative_risk = get_relative_risk_data(builder, self.risk, self.target)
```
Comment (rmudambi) [inline src/vivarium_public_health/risks/effect.py:106]: Remove this commented code.

**Review by rmudambi — CHANGES_REQUESTED** [review]
Let's talk after standup about the data loading functions.

File: src/vivarium_public_health/risks/effect.py [hunk 3]
```diff
@@ -40,34 +43,37 @@ class RiskEffect(Component):
...
         this component.
         """
         return {
-            f"effect_of_{self.risk.name}_on_{self.target.name}": {
-                self.target.measure: self.CONFIGURATION_DEFAULTS["effect_of_risk_on_target"][
-                    "measure"
-                ]
+            self.name: {
+                "data_sources": {
+                    "relative_risk": "self::get_relative_risk_source",
```
Comment (albrja) [inline src/vivarium_public_health/risks/effect.py:67]: @rmudambi The way we have to call these methods in build lookup table makes me think this is either not configurable or I should have parsed them from the configuration when we call them in that method

File: src/vivarium_public_health/risks/data_transformations.py [hunk 4]
```diff
@@ -521,6 +537,7 @@ def validate_relative_risk_rebin_source(
 
 
 def validate_rebin_source(builder, risk: EntityString, data: pd.DataFrame):
+    # TODO: when this is Falsey, it now throws a ConfigurationError
```
Comment (rmudambi) [inline src/vivarium_public_health/risks/data_transformations.py:540]: What is this todo?

File: src/vivarium_public_health/risks/effect.py (hunk 3 above)
Comment (rmudambi) [reply src/vivarium_public_health/risks/effect.py:67]: I'm not sure I understand this comment. This is configurable, as you could write a different function to get the data, or you could just provide an artifact key to load the data directly, if that data was already in the correct format.

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_public_health/risks/data_transformations.py (hunk 4 above)
Comment (albrja) [reply src/vivarium_public_health/risks/data_transformations.py:540]: Just a reminder for our discussion yesterday

File: src/vivarium_public_health/risks/data_transformations.py [hunk 5]
```diff
@@ -228,16 +231,18 @@ def get_relative_risk_data(builder, risk: EntityString, target: TargetString):
...
 
 def load_relative_risk_data(
-    builder, risk: EntityString, target: TargetString, source_type: str
-):
-    relative_risk_source = builder.configuration[f"effect_of_{risk.name}_on_{target.name}"][
-        target.measure
-    ]
+    builder: Builder, risk: EntityString, target: TargetString, source_type: str
+) -> pd.DataFrame:
+    from vivarium_public_health.risks import RiskEffect
```
Comment (patricktnast) [inline src/vivarium_public_health/risks/data_transformations.py:242]: Is there a reason why you're doing this import here and on ln 470 inline and not at the top of the file?

File: src/vivarium_public_health/risks/data_transformations.py [hunk 6]
```diff
@@ -393,31 +398,35 @@ def exposure_effect(rates, rr: pd.DataFrame) -> pd.Series:
...
-        key_cols = ["sex", "age_start", "age_end", "year_start", "year_end"]
-        exposure_data = get_exposure_data(builder, risk).set_index(key_cols)
-        relative_risk_data = get_relative_risk_data(builder, risk, target).set_index(key_cols)
+        exposure_data, exposure_value_cols = get_exposure_data(builder, risk)
+        relative_risk_data, rr_value_cols = get_relative_risk_data(builder, risk, target)
+        if set(exposure_value_cols) != set(rr_value_cols):
+            error_msg = "Exposure and relative risk value columns must match. "
+            missing_rr_cols = set(exposure_value_cols).difference(set(rr_value_cols))
+            if missing_rr_cols:
+                error_msg = error_msg + f"Missing relative risk columns: {missing_rr_cols}. "
```
Comment (patricktnast) [inline src/vivarium_public_health/risks/data_transformations.py:418]: nit: you can do `+=` here

File: src/vivarium_public_health/risks/data_transformations.py [hunk 7]
```diff
@@ -393,31 +398,35 @@ def exposure_effect(rates, rr: pd.DataFrame) -> pd.Series:
...
         )
         paf_data = paf_data[correct_target].drop(
             columns=["affected_entity", "affected_measure"]
         )
     else:
-        key_cols = ["sex", "age_start", "age_end", "year_start", "year_end"]
-        exposure_data = get_exposure_data(builder, risk).set_index(key_cols)
-        relative_risk_data = get_relative_risk_data(builder, risk, target).set_index(key_cols)
+        exposure_data, exposure_value_cols = get_exposure_data(builder, risk)
+        relative_risk_data, rr_value_cols = get_relative_risk_data(builder, risk, target)
```
Comment (patricktnast) [inline src/vivarium_public_health/risks/data_transformations.py:413]: nit: maybe wrap `exposure_val_cols` and `rr_value_cols` in `set` up front so you don't have to do it every time.

**Review by patricktnast — APPROVED** [review]

**Pull request merged**